tests/lib/nested.sh

```diff
@@ -127,72 +128,73 @@ refresh_to_new_core(){
     fi
 }
 
-create_nested_core_vm(){
-    local UBUNTU_IMAGE
-    UBUNTU_IMAGE=$(command -v ubuntu-image)
-
-    # create ubuntu-core image
-    local EXTRA_SNAPS=""
-    if [ -d "${PWD}/extra-snaps" ] && [ "$(find "${PWD}/extra-snaps/" -type f -name "*.snap" | wc -l)" -gt 0 ]; then
-        EXTRA_SNAPS="--extra-snaps ${PWD}/extra-snaps/*.snap"
-    fi
-    mkdir -p "$WORK_DIR"
-
-    local NESTED_MODEL=""
-    case "$SPREAD_SYSTEM" in
-    ubuntu-16.04-64)
-        NESTED_MODEL="$TESTSLIB/assertions/nested-amd64.model"
-        ;;
-    ubuntu-18.04-64)
-        NESTED_MODEL="$TESTSLIB/assertions/nested-18-amd64.model"
-        ;;
-    *)
-        echo "unsupported system"
-        exit 1
-        ;;
-    esac
+cleanup_nested_env(){
+    rm -rf "$WORK_DIR"
+}
 
-    "$UBUNTU_IMAGE" --image-size 3G "$NESTED_MODEL" \
-        --channel "$CORE_CHANNEL" \
-        --output "$WORK_DIR/ubuntu-core.img" "$EXTRA_SNAPS"
+create_nested_core_vm(){
+    mkdir -p "$WORK_DIR/image"
+    if [ ! -f "$WORK_DIR/image/ubuntu-core.img" ]; then
+        local UBUNTU_IMAGE
+        UBUNTU_IMAGE=$(command -v ubuntu-image)
+
+        # create ubuntu-core image
+        local EXTRA_SNAPS=""
+        if [ -d "${PWD}/extra-snaps" ] && [ "$(find "${PWD}/extra-snaps/" -type f -name "*.snap" | wc -l)" -gt 0 ]; then
+            EXTRA_SNAPS="--extra-snaps ${PWD}/extra-snaps/*.snap"
+        fi
 
-    create_assertions_disk
-    start_nested_core_vm
-    if wait_for_ssh; then
-        prepare_ssh
-    else
-        echo "ssh not stablished, exiting..."
-        exit 1
+        local NESTED_MODEL=""
+        case "$SPREAD_SYSTEM" in
+        ubuntu-16.04-64)
+            NESTED_MODEL="$TESTSLIB/assertions/nested-amd64.model"
+            ;;
+        ubuntu-18.04-64)
+            NESTED_MODEL="$TESTSLIB/assertions/nested-18-amd64.model"
+            ;;
+        *)
+            echo "unsupported system"
+            exit 1
+            ;;
+        esac
+
+        "$UBUNTU_IMAGE" --image-size 3G "$NESTED_MODEL" \
+            --channel "$CORE_CHANNEL" \
+            --output "$WORK_DIR/image/ubuntu-core.img" "$EXTRA_SNAPS"
+
+        create_assertions_disk
     fi
 }
 
 start_nested_core_vm(){
-    local QEMU
+    local IMAGE QEMU
     QEMU=$(get_qemu_for_nested_vm)
-    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic \
+    # As core18 systems use to fail to start the assetion disk when using the
+    # snapshot feature, we copy the original image and use that copy to start
```

Comment:
This sounds a bit weird, snapshots should be completely transparent to the guest system, what was the error you experienced?

Reply:
Thanks for reviewing!! 
I thought the same but using snapshot I was happening that the vm was not working well with the user assertion, so the image vm was starting but the user assertion was discarded and console-conf was triggered instead. 
I was researching that isssue but couldn't find nothing useful, if you have any idea about which could be the reason please tell me.

Reply:
That's weird, I'm using snapshots locally all the time, although I'm normally passing an additional image with cloud-init config, rather than a user assertion. No issues observed with core18 images. Let me try the figure out how do generate an assertion locally and run that.

Reply:
So I built the image, and passed an assertions disk to qemu as well. I could see that snapd.autoimport.service complains about unknown account key:
```
-- Logs begin at Wed 2019-06-12 08:10:55 UTC, end at Wed 2019-06-12 08:11:38 UTC. --
Jun 12 08:10:57 localhost systemd[1]: Starting Auto import assertions from block devices...
Jun 12 08:10:57 localhost snap[781]: cmd_auto_import.go:142: error: cannot import /run/snapd/auto-import/7k87yxVV-aCE6l1RYK6z11i1HPy0v57GH4arAJcD_50fTfjmKcnbOVcJlWGdG9eo.assert: assert failed: cannot find account-key (0u-IKZyVlTxwiBVOv7oT1FNKdfkKJyOSwbEWJPurlvtaGZxX6VrMHSESnIPeaNMC)
Jun 12 08:10:57 localhost systemd[1]: Started Auto import assertions from block devices.
```
I've build snapd with test keys enabled, but maybe I should have used a different model?

Reply:
@the account-key has to be the same you used to sign the model. So you will need to create the new model with your account-key and sign the model doing: 
snap sign  -k <key-name>  < <name>.model.json > <name>.model